ignite/distributed/launcher.py

```diff
@@ -344,3 +358,16 @@ def __exit__(self, *args: Any, **kwargs: Any) -> None:
                 f"Finalized processing group with backend: '{self.backend}'"
             )
             idist.finalize()
+
+
+@dataclass
+class RankProcessFirst:
```

Comment:
Let's make the following API:
```python
with idist.one_process_first(rank=1):
    do_something()
```
First, about the name, I think maybe `one_process_first` is better than `rank_process_first` ?
So, let's introduce a function that returns a private class (`RankProcessFirst` like you coded) that implements a context manager interface. Or maybe we could use directly python built-in [contextlib](https://docs.python.org/3.8/library/contextlib.html) taking into account that we want to support python versions 3.8 - 3.11 (and without breaking 3.7)